test/e2e/dataplane/tcp_pod_to_service.go

```diff
@@ -14,45 +14,54 @@ var _ = Describe("[dataplane] Basic Pod to Service tests across clusters without
 	f := framework.NewDefaultFramework("dataplane-p2s-nd")
 
 	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters NonGW to NonGW", func() {
-		testPod2ServiceTCP(f, framework.NonGatewayNode, framework.NonGatewayNode)
+		TestPod2ServiceTCP(f, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterA)
 	})
 
 	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters GW to GW", func() {
-		testPod2ServiceTCP(f, framework.GatewayNode, framework.GatewayNode)
+		TestPod2ServiceTCP(f, framework.GatewayNode, framework.GatewayNode, framework.ClusterB, framework.ClusterA)
 	})
 
 	It("Should preserve the source IP (GW to GW node)", func() {
-		testPod2ServiceTCPIPPreservation(f, framework.GatewayNode, framework.GatewayNode)
+		TestPod2ServiceTCPIPPreservation(f, framework.GatewayNode, framework.GatewayNode)
 	})
 
 	It("Should preserve the source IP (NonGW to NonGW node)", func() {
-		testPod2ServiceTCPIPPreservation(f, framework.NonGatewayNode, framework.NonGatewayNode)
+		TestPod2ServiceTCPIPPreservation(f, framework.NonGatewayNode, framework.NonGatewayNode)
 	})
 })
 
-func testPod2ServiceTCP(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling) {
+func TestPod2ServiceTCP(f *framework.Framework, leftScheduling framework.NetworkPodScheduling, rightScheduling framework.NetworkPodScheduling, listernerPodCluster framework.ClusterIndex, connectorPodCluster framework.ClusterIndex) {
```

Comment:
Can you make these signature changes in a separate PR as I'll need it as well?

Reply:
https://github.com/submariner-io/submariner/pull/205